### PR TITLE
Added ability to slice Reader object using lists and arrays of integers

### PR DESCRIPTION
**Pull request by richardjgowers**
Fixes Issue #437

Added tests

### CODE REVIEW

File: testsuite/MDAnalysisTests/test_reader_api.py [hunk 1]
```diff
@@ -114,7 +115,7 @@ class _Multi(object):
     n_frames = 10
     n_atoms = 10
     readerclass = AmazingMultiFrameReader
-    reference = [i for i in range(10)]
+    reference = np.array([i for i in range(10)])
```
Comment (dotsdl) [inline testsuite/MDAnalysisTests/test_reader_api.py]: A nitpick, but perhaps instead `np.arange(10)`?

File: testsuite/MDAnalysisTests/test_reader_api.py [hunk 2]
```diff
@@ -175,6 +176,24 @@ def sl():
...
+    def test_list_TE(self):
+        def sl():
+            return list(self.reader[[0, 'a', 5, 6]])
+        assert_raises(TypeError, sl)
+
+    def test_array_TE(self):
+        def sl():
+            return list(self.reader[np.array([1.2, 3.4, 5.6])])
+        assert_raises(TypeError, sl)
+
```
Comment (dotsdl) [inline testsuite/MDAnalysisTests/test_reader_api.py]: Can you add a test for each of:

1. Repeated indices?
2. Out-of-order indices?

File: testsuite/MDAnalysisTests/test_reader_api.py (hunk 1 above)
Comment (orbeckst) [inline testsuite/MDAnalysisTests/test_reader_api.py]: +1

**Comment by richardjgowers** [comment]
@dotsdl poke

File: testsuite/MDAnalysisTests/test_coordinates.py [hunk 3]
```diff
@@ -1720,6 +1720,22 @@ def test_slice_dcd(self):
...
+
+    def test_list_reverse_trajectory(self):
+        frames = [ts.frame for ts in self.dcd[[0, 4, 2, 3, 0, 1]]]
+        assert_equal(frames, [0, 4, 2, 3, 0, 1])
+
+    def test_list_reverse_trajectory(self):
+        frames = [ts.frame for ts in self.dcd[[0, 0, 1, 1, 2, 1, 1]]]
+        assert_equal(frames, [0, 0, 1, 1, 2, 1, 1])
+
     def test_reverse_dcd(self):
```
Comment (dotsdl) [inline testsuite/MDAnalysisTests/test_coordinates.py]: Are the tests in `test_coordinates.py` redundant, given that practically the same tests exist in `test_reader_api.py`? I'm not sure I understand the reason for having both.

**Comment by dotsdl** [comment]
Are the tests in test_coordinates.py redundant, given that practically the same tests exist in test_reader_api.py? I'm not sure I understand the reason for having both.

**Comment by richardjgowers** [comment]
I think ideally I want all Readers to go through the API tests (test_coordinates is a mess).  Currently reader_api only tests a dummy Reader, so I put in the DCD tests so it does a "real" Reader too

So yeah they're superfluous for now

**Comment by dotsdl** [comment]
Understood. Well, it can't hurt to have them at the moment, and I agree with the rationale. We will eventually run all Readers through the single suite of tests you've built, so we'll cut things out of `test_coordinates.py` later.

Waiting for travis to catch up after your latest merge from `develop`,  then will merge into `develop`.

**Pull request merged**